services/wireguard/connection/dns.go

```diff
@@ -25,14 +25,29 @@ import (
 	"path"
 )
 
-func setDNS(configDir, dev, dns string) error {
+// DNSManager is connection DNS configuration manager.
+type DNSManager interface {
```

Comment:
this definition should be near where it's being used, not where it's implemented

Reply:
If interface can be reused there is no point to duplicate it in different places. It just happens that there is only one implementation which is also placed in this file. There are many interfaces which you can reuse like io.Reader, io.Writer etc.

The same applies to `type HandshakeWaiter interface` which I'm using from different places. If I copy it to all used places when I also need to keep documentation in sync which is strange.